### PR TITLE
Fix quimb version checking

### LINKED ISSUE #3263: cirq.contrib.quimb fails with quimb 1.2.0
**Description of the issue**

Quimb fails with version 1.2.0.  We should either require 1.3 or (ideally) fix whatever was broken in 1.2

**How to reproduce the issue**
```
pip install quimb==1.2.0
pytest cirq/contrib/quimb/
```

<details>
test_tensor_expectation_value
...
 AttributeError: 'numpy.complex128' object has no attribute 'add_owner'
</details>

**Cirq version**
cirq from HEAD and quimb == 1.2.0

### PR DESCRIPTION
**Pull request by balopat**
Making quimb version check more robust.

In some systems the version might not be parseable as (int,int), e.g `0+unknown`. 
This PR fixes these scenarios.

Fixes #3263.

### CODE REVIEW

File: cirq/contrib/quimb/state_vector.py [hunk 1]
```diff
@@ -7,7 +7,17 @@
 
 import cirq
 
-QUIMB_VERSION = tuple(int(x) for x in quimb.__version__.split('.'))
+
+# coverage: ignore
+def _get_quimb_version():
```
Comment (dstrain115) [inline cirq/contrib/quimb/state_vector.py:12]: Should we have a short docstring (specifically what it is returning)?

**Review by dstrain115 — APPROVED** [review]
Thanks for the quick fix!

File: cirq/contrib/quimb/state_vector.py (hunk 1 above)
Comment (balopat) [reply cirq/contrib/quimb/state_vector.py:12]: added!

**Comment by mpharrigan** [comment]
Thanks!

**Pull request merged**